### PR TITLE
[SEDONA-441] Add ST_LineLocatePoint

### PR DESCRIPTION
**Pull request by prantogg**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-441. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?

Adds Vector function ST_LineLocatePoint
- Returns a double between 0 and 1, representing the location of the closest point on the LineString as a fraction of its total length.

## How was this patch tested?

- Passes new and existing tests

## Did this PR include necessary documentation updates?

- Yes, I am adding a new API. I am using the [current SNAPSHOT version number](https://github.com/prantogg/sedona/blob/f37bb46e101fac2aa1c62ace2f56818c51c17a03/pom.xml#L29) in since `vX.Y.Z` format.

### CODE REVIEW

File: docs/api/flink/Function.md [hunk 1]
```diff
@@ -1694,6 +1694,26 @@ Output:
...
+## ST_LineLocatePoint
+
+Introduction: Returns a double between 0 and 1, representing the location of the closest point on the LineString as a fraction of its total length.
+The first argument must be a LINESTRING, and the second argument is a POINT geometry.
+
+Format: `ST_LineLocatePoint(linestring: Geometry, point: Geometry)`
+
+Since: `v1.5.1`
+
+Spark SQL Example:
```
Comment (jiayuasu) [inline docs/api/flink/Function.md:1706]: Not `Spark SQL`. It is `SQL`

File: common/src/test/java/org/apache/sedona/common/FunctionsTest.java [hunk 2]
```diff
@@ -1864,4 +1864,24 @@ public void voronoiPolygons() {
...
+
+        Double actual1 = Functions.lineLocatePoint(lineString, point1);
+        Double actual2 = Functions.lineLocatePoint(lineString, point2);
+        Double actual3 = Functions.lineLocatePoint(lineString, point3);
+
+        Double expectedResult1 = 0.0;
+        Double expectedResult2 = 0.5;
+        Double expectedResult3 = 1.0;
+
+        assertEquals(expectedResult1, actual1, 0.1);
```
Comment (jiayuasu) [inline common/src/test/java/org/apache/sedona/common/FunctionsTest.java:1883]: I believe our TestBase already has a variable called `FP Tolerance`

File: flink/src/test/java/org/apache/sedona/flink/FunctionTest.java [hunk 3]
```diff
@@ -771,12 +779,6 @@ public void testRemovePointWithIndex() {
         assertEquals("LINESTRING (0 0, 2 2)", first(pointTable).getField(0).toString());
     }
 
-    @Test
```
Comment (jiayuasu) [inline flink/src/test/java/org/apache/sedona/flink/FunctionTest.java:774]: Why remove this test?

**Review by jiayuasu — CHANGES_REQUESTED** [review]

**Review by jiayuasu — APPROVED** [review]

**Comment by jpd-defra** [comment]
Thank you @prantogg  and @jiayuasu. This is great to see!

**Pull request merged**